packages/ag-charts-community/src/chart/data/dataModel.ts

```diff
@@ -14,7 +14,7 @@ export interface DataGroup {
     keys: any[];
     datumIndices: number[][];
     aggregation: any[][];
-    validScopes: Set<ScopeId> | undefined;
+    validScopes: Set<ScopeId>;
```

Comment:
Is it worth trying to keep the `| undefined` (i.e. all scopes are valid) to reduce allocations on large datasets?

Reply:
This is handled by using a single shared `Set` instance, as it's simpler for naive cases just to always have a `Set` present - we can handle performance sensitive cases as exceptions (there are already a couple).